xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -977,6 +979,30 @@ private static String serializePropertyValue(PropertyInterface property)
         }
     }
 
+    /**
+     * Serializes the value of the given XObject property. In case the property is a ComputedFieldClass, the serialized
+     * value is the computed property value.
+     * @param property an XObject property
+     * @param propertyClass the PropertyClass of that XObject proprety
+     * @param context the XWikiContext
+     * @return the String representation of the property value
+     */
+    private static String serializePropertyValue(PropertyInterface property,
+        com.xpn.xwiki.objects.classes.PropertyClass propertyClass, XWikiContext context)
+    {
+        if (propertyClass instanceof ComputedFieldClass) {
+            // TODO: the XWikiDocument needs to be explicitely set in the context, otherwise method
+            // PropertyClass.renderInContext fires a null pointer exception: bug?
+            context.setDoc(property.getObject().getOwnerDocument());
```

Comment:
It's a good practice to always reset the context document to the original value.

Reply:
Thanks for the review, I've added a context document reset in the latest commit.